### PR TITLE
expose a prop to configure container side margin

### PR DESCRIPTION
**Pull request by skeie**
Quik fix for #56

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -491,9 +497,9 @@ export default class Carousel extends Component {
     }
 
     render () {
-        const { sliderWidth, itemWidth, containerCustomStyle, contentContainerCustomStyle, enableMomentum } = this.props;
+        const { sliderWidth, itemWidth, containerCustomStyle, contentContainerCustomStyle, enableMomentum, carouselSideMargin } = this.props;
 
-        const containerSideMargin = (sliderWidth - itemWidth) / 2;
+        const containerSideMargin = carouselSideMargin || this.getContainerSideMargin(sliderWidth, itemWidth);
```
Comment (bd-arc) [inline index.js]: @skeie Hi there! Unless I'm mistaken, I think you forgot to add the method `getContainerSideMargin()` ;-)

Comment (skeie) [inline index.js]: Ops, sorry. Good catch 👍.

Should have added it now.

Comment (bd-arc) [inline index.js]: @skeie Hey! Sorry for not getting back to you sooner; I was on holidays.
If think that we should remove the method `getContainerSideMargin()` altogether. What about this?
```
const containerSideMargin = carouselSideMargin || (sliderWidth - itemWidth) / 2;
```

File: index.js [hunk 2]
```diff
@@ -102,7 +102,12 @@ export default class Carousel extends Component {
         /**
          * Fired when snapping to an item
          */
-        onSnapToItem: PropTypes.func
+        onSnapToItem: PropTypes.func,
+        /**
+         * If you want to set the side margin of the scroller
+         * This is not optimal, I may or may not work in your case
+         */
+        carouselSideMargin: PropTypes.number,
```
Comment (bd-arc) [inline index.js]: @skeie My opinion is that this prop should be renamed `carouselHorizontalPadding`, as it better describes its effect. What do you think?

**Comment by bd-arc** [comment]
@skeie Hi there! I've taken the liberty of implementing the changes I was asking for. Version 2.1.2 of the plugin should suit your needs ;-)

I still think that overriding the default padding can result in weird side effects, so please let me know if you happen to face any.

**Comment by skeie** [comment]
Sorry @bd-arc. I haven't been able to fix your comments until now. Good to hear that you took matters into their own hands 👍 .

I will let you know if we encounter any problems. Thank you - hope you enjoyed your vacation.

**Pull request closed without merging**